toolbox/bulk_processing/find_invalid_address_case_id_by_uprn.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import csv
+import os
+from pathlib import Path
+
+import requests
+from google.cloud import storage
+from requests import HTTPError
+
+from config import Config
+
+
+def generate_bulk_invalid_address_file(file_to_process):
+    address_delta_file = Path(f'invalid_addresses_{file_to_process.stem}.csv')
+    address_delta_file.write_text('case_id,reason\n')
+    error_count = 0
+
+    with open(file_to_process, encoding="utf-8") as open_file_to_process:
+        file_reader = csv.DictReader(open_file_to_process, delimiter=',')
+
+        for line_number, row in enumerate(file_reader, 1):
+            case_id_list = get_case_id_from_case_api(row['UPRN'], line_number)
+            if case_id_list:
+                for case_id in case_id_list:
+                    write_invalid_addresses_case_id_file(case_id, address_delta_file)
+            else:
+                error_count += 1
+
+        if error_count != 0:
+            os.remove(Path(f'invalid_addresses_{file_to_process.stem}.csv'))
+            exit(1)
+
+    upload_file_to_bucket(address_delta_file)
+
+
+def get_case_id_from_case_api(uprn, line_number):
+    response = requests.get(f'http://{Config.CASEAPI_HOST}:{Config.CASEAPI_PORT}/cases/uprn/{uprn}')
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        if response.status_code == 404:
+            print(f'Error 404: Cannot find the UPRN {uprn} on line {line_number}, Error {e}')
+            return False
```

Comment:
returning `False` is a bit weird isn't it? I would expect it to return a tuple, or `None`

Reply:
Agreed, `None` or an empty list would be preferable.